### PR TITLE
Update changelog guide and Giles configuration

### PR DESCRIPTION
**Pull request by namurphy**
This PR makes the following changes:
 - Updated the changelog guide, in particular to fix an old link
 - Updated the configuration for Giles, which is what we use to check changelog entries. In particular, I updated the error messages including adding a link to the changelog guide.

I didn't include a changelog entry for the Giles configuration update, since it's not user facing and a pretty minor change anyway.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -208,14 +208,14 @@ enabled = true
...
+changelog_missing = "Missing changelog entry (see `changelog/README.rst`)"
+changelog_missing_long = "This pull request needs a changelog entry at `changelog/NUMBER.TYPE.rst`, where `NUMBER` is the pull request number and `TYPE` is one of `feature`, `trivial`, `doc`, `bugfix`, `breaking`, or `removal`.\n\nFor details, see: https://docs.plasmapy.org/en/latest/contributing/changelog_guide.html#adding-a-changelog-entry"
 verify_pr_number = true
-number_incorrect = "Incorrect changelog entry number (match PR!)"
-number_incorrect_long = "The changelog entry's number does not match this pull request's number."
-type_incorrect = "Incorrect changelog entry type (see list in changelog README)"
-type_incorrect_long = "The changelog entry for this PR must have one of the types (as in NUMBER.TYPE.rst) as described in the changelog README)."
+number_incorrect = "Changelog entry number ≠ PR number (see `changelog/README.rst`)"
+number_incorrect_long = "The changelog entry number does not match this pull request's number.\n\nWhen purposefully editing the changelog entry for a different pull request, set the 'No changelog entry needed' label to ignore this check."
```
Comment (namurphy) [inline pyproject.toml:216]: After this is merged, I'll have to check to make sure that the `\n` gets interpreted as a line break.

**Review by pheuer — APPROVED** [review]
Looks good - reminder to check the links (and the /n) after merge!

**Comment by namurphy** [comment]
Thank you again!

**Comment by namurphy** [comment]
Just checked another PR without a changelog, and the `\n` and link are working fine.  Thanks for the reminder — I probably would have forgotten otherwise! 🙃

**Pull request merged**